### DISCUSSION TITLE
Docker - Rpi - 404 on ldc2-1.16.0-linux-.tar.xz

### DISCUSSION THREAD

**Discussion by thomaswwp** [abraunegg/onedrive#1331] [Q&A] [answered]
```
Step 4/11 : RUN wget https://github.com/ldc-developers/ldc/releases/download/v1.16.0/ldc2-1.16.0-linux-${ARCH}.tar.xz &&     tar -xvf ldc2-1.16.0-linux-${ARCH}.tar.xz
 ---> Running in eaddb3bf4ca7
--2021-03-10 09:35:16--  https://github.com/ldc-developers/ldc/releases/download/v1.16.0/ldc2-1.16.0-linux-.tar.xz
Resolving github.com (github.com)... 140.82.121.4
Connecting to github.com (github.com)|140.82.121.4|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2021-03-10 09:35:17 ERROR 404: Not Found.

The command '/bin/sh -c wget https://github.com/ldc-developers/ldc/releases/download/v1.16.0/ldc2-1.16.0-linux-${ARCH}.tar.xz &&     tar -xvf ldc2-1.16.0-linux-${ARCH}.tar.xz' returned a non-zero code: 8
```

I am a hobbyist, not an expert and trying to get this excellent docker version running on my RPi. It is a slow learning process but this one has stumped me. I am guessing that I can edit the script somehow to put in a correct link to a correct ldc but have no idea where to start.

Later: I am wondering if the ${ARCH} should produce something like "armhf" but just guessing. I tried `export ARCH="armhf"` but it made no difference.

**Comment by thomaswwp** [comment, marked as answer]
After making that change to the environment variable this worked: 

`docker build --build-arg ARCH=${ARCH} . -t local-onedrive-rpi -f contrib/docker/Dockerfile-rpi`

I guess I could have just used ARCH="armhf".

**Reply by abraunegg** [reply]
The Dockerfile for RPI has the following:
```
ARG ARCH armhf
```
By default, this should be using `armhf` to complete the command sequence to download the right files - but it looks like this is failing.

Valid entries for this should be:
* armhf
* aarch64

What I will do is update this Dockerfile specifically for ARMHF, and add another for AARCH64

This issue is being caused by: https://github.com/abraunegg/onedrive/pull/1259

**Reply by abraunegg** [reply]
@thomaswwp 
Please can you test the following PR which updates the Docker files:
```
git clone https://github.com/abraunegg/onedrive.git
cd onedrive
git fetch origin pull/1335/head:pr1335
git checkout pr1335
```

**To build a Docker image for ARMHF (Raspberry Pi)**
Compatible with:
*    Raspberry Pi
*    Raspberry Pi 2
*    Raspberry Pi Zero
*    Raspberry Pi 3
```
docker build . -t local-onedrive-rpi -f contrib/docker/Dockerfile-armhf
```

**To build a Docker image for AARCH64 (Raspberry Pi)**
Compatible with:
*    Raspberry Pi 4 if not running a ARMHF based Linux host OS
```
docker build . -t local-onedrive-rpi -f contrib/docker/Dockerfile-aarch64
```

**Reply by thomaswwp** [reply]
Sorry for the short delay. 

I am a RPi 4 running minimal Raspbian which I think is armhf, so used the first command (maybe add to that lisr "Raspberry Pi 4 running armhf image").

It all ran as expected with the above git commands and
`docker build . -t local-onedrive-rpi -f contrib/docker/Dockerfile-armhf`

Could I ask that for inexperienced people like me you add to the custom builds at the end of Docker.md that they need to put the name of the image (e.g. local-onedrive-rpi) at the end of the `docker run` command instead of `driveone/onedrive:latest` (I think, from memory, that is how I got it working)

**Reply by thomaswwp** [reply]
Sorry for the short delay. 

I am a RPi 4 running minimal Raspbian which I think is armhf, so used the first command (maybe add to that lisr "Raspberry Pi 4 running armhf image").

It all ran as expected with the above git commands and
`docker build . -t local-onedrive-rpi -f contrib/docker/Dockerfile-armhf`

Could I ask that for inexperienced people like me you add to the custom builds at the end of Docker.md that they need to put the name of the image (e.g. local-onedrive-rpi) at the end of the `docker run` command instead of `driveone/onedrive:latest` (I think, from memory, that is how I got it working)

**Reply by thomaswwp** [reply]
Sorry for the short delay. 

I am a RPi 4 running minimal Raspbian which I think is armhf, so used the first command (maybe add to that lisr "Raspberry Pi 4 running armhf image").

It all ran as expected with the above git commands and
`docker build . -t local-onedrive-rpi -f contrib/docker/Dockerfile-armhf`

Could I ask that for inexperienced people like me you add to the custom builds at the end of Docker.md that they need to put the name of the image (e.g. local-onedrive-rpi) at the end of the `docker run` command instead of `driveone/onedrive:latest` (I think, from memory, that is how I got it working)
